src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,36 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const status = useAppSelector(state => state.filter.status);
+
+  const handleSelection = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterSlice.actions.setStatus(event.target.value));
```

Comment:
Ensure that the `event.target.value` matches the expected type for the `setStatus` action in the `filterSlice`. If `Status` is a TypeScript enum or a specific type, you may need to cast or validate the value before dispatching.